### ISSUE TITLE
Using spymemcached-2.11.7

### ISSUE THREAD

**Issue by tuomash** [magro/memcached-session-manager#318]
We have been using memcached-session-manager-1.8.3.jar, memcached-session-manager-tc7-1.8.3.jar and spymemcached-2.11.1.jar successfully in our staging environment for a long while, but when we setup another environment using the same library versions (and same Tomcat and Java versions) we ran into some issues.

It seems that after a while (a day or so) spymemcached-2.11.1 gets deadlocked like so:

"Memcached IO over {MemcachedConnection to /x.x.x.x:11211 /x.x.x.x:11211}" prio=5 tid=109 BLOCKED
 at net.spy.memcached.internal.AbstractListenableFuture.notifyListeners(AbstractListenableFuture.java:144)
 at net.spy.memcached.internal.AbstractListenableFuture.notifyListeners(AbstractListenableFuture.java:130)
 at net.spy.memcached.internal.GetFuture.signalComplete(GetFuture.java:109)
 at net.spy.memcached.MemcachedClient$5.complete(MemcachedClient.java:1041)
       Local Variable: net.spy.memcached.MemcachedClient$5#84
 at net.spy.memcached.protocol.BaseOperationImpl.transitionState(BaseOperationImpl.java:167)
       Local Variable: net.spy.memcached.ops.OperationState#4
 at net.spy.memcached.protocol.ascii.BaseGetOpImpl.handleLine(BaseGetOpImpl.java:99)
       Local Variable: java.lang.String#410564
 at net.spy.memcached.protocol.ascii.OperationImpl.readFromBuffer(OperationImpl.java:154)
 at net.spy.memcached.MemcachedConnection.readBufferAndLogMetrics(MemcachedConnection.java:825)
 at net.spy.memcached.MemcachedConnection.handleReads(MemcachedConnection.java:804)
       Local Variable: sun.nio.ch.SocketChannelImpl#3Local Variable: java.nio.DirectByteBuffer#33
 at net.spy.memcached.MemcachedConnection.handleReadsAndWrites(MemcachedConnection.java:684)
       Local Variable: sun.nio.ch.SelectionKeyImpl#9
 at net.spy.memcached.MemcachedConnection.handleIO(MemcachedConnection.java:647)
       Local Variable: net.spy.memcached.protocol.ascii.AsciiMemcachedNodeImpl#1
 at net.spy.memcached.MemcachedConnection.handleIO(MemcachedConnection.java:418)
       Local Variable: sun.nio.ch.Util$2#1Local Variable: java.util.HashMap$KeyIterator#37265
 at net.spy.memcached.MemcachedConnection.run(MemcachedConnection.java:1401)

This shows in logs as constant timeouts to the memcached servers, but when I checked with netstat I saw that the Java process had established connections to the memcached servers normally. The only way to recover this situation is to restart Tomcat.

This issue seems to be fixed in spymemcached 2.11.2: https://github.com/couchbase/spymemcached/releases/tag/2.11.2.

So my question is, is it safe to use the latest spymemcached 2.11 version, which is 2.11.7, with memcached-session-manager version 1.8.3? The SetupdAndConfiguration page suggests using 2.11.1 but has it been updated in a while?

**Comment by tuomash** [comment]
We have been now running spymemcached 2.11.7 with memcached-session-manager-1.8.3 without any problems, so I think this issue can be closed.

**Comment by jeffbenedict83** [comment]
This fixed our problem as well, but we have no idea the root cause of the issue was.  We updated to 2.11.1 back in October/November of 2016, but didn't see any deadlock issues until March 12, 2017, and they occurred multiple times a day until we updated the spymemcached jar to 2.12.2.

We couldn't reproduce this reliably under load in non production either.  We would load test with 2.11.1, and would deadlock, then load test again with 2.11.1 and wouldn't deadlock.  

We could use some help with root cause.  Any help would be greatly appreciated.

Thanks in advance!
